### DISCUSSION TITLE
Using plustoolkit with hybrid Linux/Windows setup

### DISCUSSION THREAD

**Discussion by kenavolic** [PlusToolkit/PlusLib#849] [Q&A] [answered]
Hi,

First of all, thanks for your impressive work with plustoolkit!

I am wondering if the current capabilities of the framework could solve the setup issue we have in our new project. Here is a setup:
- a Windows system with a Telemed driver running
- a Linux system with an optical tracker running

For the acquisition step, I was thinking of:
1- a PlusServer on the windows side to forward frame via OpenIGTLink
2- a custom driver to retrieve tracker data on the Linux side
3- a custom OpenIGTLink client on the Linux side to retrieve the streamed data (as we don't use Slicer)

For the second point, I was wondering if this is possible to use a local PlusServer on the Linux side with a "OpenIGTLinkVideo" to plug to the OpenIGTLink stream of the Windows PlusSever mixed with an optical driver device running locally? Moreover, could you confirm that such a setup would be needed for the calibration of an utlrasound system and a tracker system that do not reside on the same computer?

Once I know it is feasible or not, I will have to deal with the issue of syncing both data stream...

If you can redirect me to any good documentation or example of such a setup, it would be great!

Thanks for your help,
Best Regards,
Charles

**Comment by Sunderlandkyl** [comment, marked as answer]
Yes, you can collect images using PlusServer running on windows and combine them with tracking data collected from PlusServer on Linux. As you mentioned, this can be done with the "OpenIGTLinkVideo" device. You can combine the two data streams on the Linux server using a "VirtualMixer" device.

http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceOpenIGTLinkVideo.html
http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceVirtualMixer.html

**Reply by kenavolic** [reply]
Thanks, I'll give it a try.
